src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/ComWrappersSupport.cs

```diff
@@ -0,0 +1,187 @@
+            var interfaces = obj.GetType().GetInterfaces();
+            foreach (var iface in interfaces)
+            {
+                if (Projections.IsTypeWindowsRuntimeType(iface))
+                {
+                    var ifaceAbiType = iface.FindHelperType();
+                    entries.Add(new ComInterfaceEntry
+                    {
+                        IID = GuidGenerator.GetIID(ifaceAbiType),
+                        Vtable = (IntPtr)ifaceAbiType.FindVftblType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                    });
+                }
+
+                if (iface.IsConstructedGenericType
+                    && Projections.TryGetCompatibleWindowsRuntimeTypeForVariantType(iface, out var compatibleIface))
+                {
+                    var compatibleIfaceAbiType = compatibleIface.FindHelperType();
+                    entries.Add(new ComInterfaceEntry
+                    {
+                        IID = GuidGenerator.GetIID(compatibleIfaceAbiType),
+                        Vtable = (IntPtr)compatibleIfaceAbiType.FindVftblType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                    });
+                }
+            }
+
+            if (obj is Delegate)
+            {
+                entries.Add(new ComInterfaceEntry
+                {
+                    IID = GuidGenerator.GetIID(obj.GetType()),
+                    Vtable = (IntPtr)obj.GetType().GetHelperType().GetField("AbiToProjectionVftablePtr", BindingFlags.Public | BindingFlags.Static).GetValue(null)
+                });
+            }
+            // Add IAgileObject to all CCWs
+            entries.Add(new ComInterfaceEntry
+            {
+                IID = IID_IAgileObject,
+                Vtable = IUnknownVftbl.AbiToProjectionVftblPtr
+            });
+            return entries;
+        }
+
+        public static (InspectableInfo inspectableInfo, List<ComInterfaceEntry> interfaceTableEntries) PregenerateNativeTypeInformation(object obj)
+        {
+            var interfaceTableEntries = GetInterfaceTableEntries(obj);
+            var iids = new Guid[interfaceTableEntries.Count];
+            for (int i = 0; i < interfaceTableEntries.Count; i++)
+            {
+                iids[i] = interfaceTableEntries[i].IID;
+            }
+
+            Type type = obj.GetType();
+
+            if (type.FullName.StartsWith("ABI."))
+            {
+                type = Projections.FindCustomPublicTypeForAbiType(type) ?? type.Assembly.GetType(type.FullName.Substring("ABI.".Length)) ?? type;
+            }
+
+            return (
+                new InspectableInfo(type, iids),
+                interfaceTableEntries);
+        }
+
+        internal class InspectableInfo
+        {
+            public Guid[] IIDs { get; }
+            // WPF's usage of WinRT doesn't create any ccws, so we can return an empty string.
```

Comment:
What is the `ccws`?

Reply:
Com callable wrapper. Basically WPF doesn't pass any managed objects to native, so we don't need full support here.

Reply:
@jkoritzinsky Thank you.

Reply:
If this is generated code how come there are WPF spec fix comments here? I'm confused....

Reply:
I've made a small number of functional changes here to remove functionality (this allowed me to remove 100s of lines of code)

Reply:
I've tried to make as few changes as possible. The only changes I have made have been to enable removing significant amounts of code since this might be going into preview 6.